### PR TITLE
add integration-image for integration test in testflight

### PR DESCRIPTION
**Pull request by kcmannem**
fixes concourse/concourse#4878

Signed-off-by: Bin Ju <bju@pivotal.io>
Co-authored-by: Krishna Mannem <kmannem@pivotal.io>

### CODE REVIEW

File: pipelines/concourse.yml [hunk 1]
```diff
@@ -128,6 +128,30 @@ jobs:
         mode: normal
         alert_type: broke
 
+- name: build-integration-image
```
Comment (deniseyu) [inline pipelines/concourse.yml]: could we call this `build-containerd-integration-image` or something similar. Someone who rolls onto the team might find it extremely confusing that this isn't the image used for our main integration tests

Comment (cirocosta) [reply pipelines/concourse.yml]: if I understood correctly, it's not *only* for testing `containerd` - it'd be used for the whole set of testflight tests, *as well as* `containerd`'s integration ones.

is that right, @kcmannem?

thx!

File: dockerfiles/integration/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Not intended to be packaged or used for production
+
+#FROM concourse/golang-builder
+FROM concourse/unit
+
+RUN mkdir -p /go/src/github.com/concourse/ /usr/local/concourse/bin/
+ENV PATH=/usr/local/concourse/bin:$PATH
+ENV GO111MODULE=on
+
+COPY start.sh /usr/local/concourse/bin/start.sh
```
Comment (cirocosta) [inline dockerfiles/integration/Dockerfile]: where is this `start.sh` coming from? 🤔  

```
 ~/workspace/ci $ git status
On branch containerd-integration
Your branch is up to date with 'origin/containerd-integration'.

nothing to commit, working tree clean
 ~/workspace/ci $ find . -name "start.sh"
```

File: dockerfiles/integration/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# This Dockerfile is only for running integration tests that depend on a running containerd process
```
Comment (cirocosta) [inline dockerfiles/integration/Dockerfile]: nit: from what I understood, it's supposed to not only running `integration` tests, but also `testflight`. 

```suggestion
 # This Dockerfile is only for running testflight and other integration tests
```

File: dockerfiles/integration/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+# Not intended to be packaged or used for production
+
+#FROM concourse/golang-builder
```
Comment (cirocosta) [inline dockerfiles/integration/Dockerfile]: nit: leftovers? 😁

File: pipelines/concourse.yml (hunk 1 above)
Comment (pivotal-bin-ju) [reply pipelines/concourse.yml]: @cirocosta you're right! It is the easiest way we could run both existed test flights and `containerd` together maybe. Any concern on this?

Comment (pivotal-bin-ju) [reply pipelines/concourse.yml]: @deniseyu good point! But I am worrying whether it is going to confuse people after renaming it to that name, while it is used for both for the whole test flights.
What about `build-containerd-enabled-integration-image`?

Comment (cirocosta) [reply pipelines/concourse.yml]: > @cirocosta [...]  Any concern on this?

ooh, no concerns! I was just raising the "why" I think this was not named after `*-containerd-*` 😁

**Comment by cirocosta** [comment]
Hey @pivotal-bin-ju @kcmannem , I just updated the pipeline so that we can `fly set-pipeline` it.

Could take a quick look at the comments above though? Specifically with regards to `start.sh`.

Thanks!

File: dockerfiles/integration/Dockerfile (hunk 4 above)
Comment (cirocosta) [reply dockerfiles/integration/Dockerfile]:
```suggestion

```

**Comment by cirocosta** [comment]
It got me wondering whether we really need to have a separate image for this - we use `unit` (which sure, could be renamed to something better) for packing all of the possibly necessary pieces of extra software needed for testing in various conditions, so, why not include containerd there? we already include `kubectl`, `helm`, etc ...

 I think the benefit of having it all in the same place is that it just becomes simpler to keep track of where those dependencies are - have a problem with a dep? just go straight to `unit`.

wdyt of having it there?

thanks!

**Comment by cirocosta** [comment]
Hey,

I went forward with the changes I described in https://github.com/concourse/ci/pull/246#issuecomment-571782505, building upon the contributions here, so I'll go ahead w/ closing this one.

Thank you all!

**Pull request closed without merging**